config/initializers/0_clowder-config.rb

```diff
@@ -19,13 +19,25 @@
   redis_url = "#{config.dig('inMemoryDb', 'hostname')}:#{config.dig('inMemoryDb', 'port')}"
   redis_password = config.dig('inMemoryDb', 'password')
 
+  # Kafka
+  first_kafka_server_config = config.kafka.brokers[0]
+
+  kafka_server_config = {
+    brokers: "#{first_kafka_server_config.dig('hostname')}:#{first_kafka_server_config.dig('port')}",
+    security_protocol: first_kafka_server_config.dig('authtype')
+  }
+
+  if kafka_server_config['security_protocol'] == 'sasl_ssl'
+    kafkaCaFile = File.new('tmp/kafkaCa', 'wt')
```

Comment:
Please use [`Tempfile`](https://ruby-doc.org/stdlib-3.0.0/libdoc/tempfile/rdoc/Tempfile.html). It would put the file in the right location with correct permissions.

Reply:
Yeah that makes perfect sense ! I have a question though , I'm following the same approach as in [here](https://github.com/RedHatInsights/compliance-backend/blob/cea75b77678f2cc8699cf1a679337a0be94ff9d1/config/database.yml#L5) , also note I'm using the relative path to our project's tmp dir. 

Are you suggesting to switch it to the system default's TMPDIR ? (likely `/tmp`) or to use Tempdir as an interface and keep using the local tmp dir ?

Reply:
I defaulted to whatever the TMPDIR is set and declined even to define a name (which is irrelevant), I only had to define the `mode` to ensure I have write permissions and that it was opened in text mode.

Reply:
Yeah, keep the defaults of Tempfile.